src/probability/martingale.lean

```diff
@@ -353,6 +386,141 @@ end
 
 end submartingale
 
+-- We now prove the converse of the optional stopping theorem
+
+section indicator_stopping_time
+
+/-- Auxilliary definition for `submartingale_of_expected_stopped_value_mono`.  -/
+noncomputable def indicator_stopping_time (i j : ℕ) (s : set α) : α → ℕ :=
+s.indicator i + sᶜ.indicator j
+
+variables {i j n : ℕ} {s : set α}
+
+lemma indicator_stopping_time_range (i j : ℕ) (s : set α) (x : α) :
```

Comment:
I think a better lemma might be
```suggestion
lemma indicator_stopping_time_eq_ite (i j : ℕ) (s : set α) (x : α) :
   indicator_stopping_time i j s x = if x ∈ s then i else j := sorry
```
I think this might be better as a lemma with `indicator_stopping_time` unfolded, so that it's just about set.indicator.

Reply:
I've added your lemma for general `mul_indicators`